### ISSUE TITLE
Access to the request's headers

### ISSUE THREAD

**Issue by NatoBoram** [jdalrymple/gitbeaker#3411]
**Description**

When using GitLab's API, it sends back some rate limit information.

```yaml
ratelimit-observed: 1
ratelimit-remaining: 1999
ratelimit-reset: 1694791726
ratelimit-resettime: Fri, 15 Sep 2023 15:28:46 GMT
ratelimit-limit: 2000
```

It would be very useful to have a way to obtain this information.

**Proposal**

To avoid breaking all currently existing schema and because rate limit fluctuates often over time, it would be useful for each request to store the latest rate limit inside the client itself so it's easily accessible.

```ts
export const client = new Gitlab({
	token: GITLAB_TOKEN,
})

interface RateLimit {
	/** Number of requests made in the current rate limit window */
	readonly observed: number

	/** Number of requests remaining in the current rate limit window */
	readonly remaining: number

	/** Time at which the current rate limit window resets */
	readonly reset: Date

	/** US-formatted date string at which the current rate limit window resets
	 *
	 * @example Fri, 15 Sep 2023 15:28:46 GMT
	 */
	readonly resettime: string

	/** Maximum number of requests allowed in the current rate limit window */
	readonly limit: number
}

const rateLimit: RateLimit = client.rateLimit
```

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by jdalrymple** [comment]
Hmm, this is returned back after each request ? I could include it in the expanded payload 

```
data: [
...
],
rateLimitInfo: //The update ratelimit information
paginationInfo: {
  next: 4,
  current: 2,
  previous: 1,
  perPage: 3,
}
```

**Comment by jdalrymple** [comment]
I realised this is already possible, just pass the showExpanded optional argument in your method call and the returned payload will include the headers and thus the rate-limit information from gitlab.

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`39.18.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/39.18.0) :rocket:
